aces/idt/core/constants.py

```diff
@@ -0,0 +1,412 @@
+"""Module which holds constants for the project
+
+"""
+from typing import ClassVar
+
+import colour
+import numpy as np
+
+from .structures import IDTMetaData
+
+
+class DataFolderStructure:
+    """Constants for the folder names which make up the data structure"""
+
+    DATA: ClassVar[str] = "data"
+    COLOUR_CHECKER: ClassVar[str] = "colour_checker"
+    GREY_CARD: ClassVar[str] = "grey_card"
+
+
+class UITypes:
+    """Constants for the UI categories"""
+
+    INT_FIELD: ClassVar[str] = "IntField"
+    STRING_FIELD: ClassVar[str] = "StringField"
+    OPTIONS_FIELD: ClassVar[str] = "OptionsField"
+    VECTOR3_FIELD: ClassVar[str] = "Vector3Field"
+    FOLDER_STRUCTURE: ClassVar[str] = "FOLDER_STRUCTURE"
+    BOOLEAN_FIELD: ClassVar[str] = "BooleanField"
+    ARRAY_FIELD: ClassVar[str] = "ArrayField"
+    MAP_FIELD: ClassVar[str] = "MapField"
+
+
+class UICategories:
+    """Constants for the UI categories"""
+
+    ADVANCED: ClassVar[str] = "Advanced"
+    STANDARD: ClassVar[str] = "Standard"
+    HIDDEN: ClassVar[str] = "HIDDEN"
+
+
+class LUTSize:
+    """Constants for the LUT sizes"""
+
+    LUT_1024: ClassVar[int] = 1024
+    LUT_2048: ClassVar[int] = 2048
+    LUT_4096: ClassVar[int] = 4096
+    LUT_8192: ClassVar[int] = 8192
+    LUT_16384: ClassVar[int] = 16384
+    LUT_32768: ClassVar[int] = 32768
+    LUT_65536: ClassVar[int] = 65536
+    DEFAULT: ClassVar[int] = LUT_1024
+    ALL: ClassVar[tuple[int, ...]] = (
+        LUT_1024,
+        LUT_2048,
+        LUT_4096,
+        LUT_8192,
+        LUT_16384,
+        LUT_32768,
+        LUT_65536,
+    )
+
+
+class RGBDisplayColourspace:
+    """Constants for the RGB display colourspaces"""
+
+    SRGB: ClassVar[str] = "sRGB"
+    DCI_P3: ClassVar[str] = "Display P3"
+    DEFAULT: ClassVar[str] = SRGB
+    ALL: ClassVar[tuple[str, ...]] = (SRGB, DCI_P3)
+
+
+class CAT:
+    """Constants for the CATs"""
+
+    DEFAULT: ClassVar[str] = "CAT02"
+
+    @classmethod
+    @property
+    def ALL(cls):
+        """
+
+        Returns A list of all the CATs
+        -------
```

Comment:
**Pedantry**

We are not building docs atm but pretty sure that Sphinx would be sad here.

Reply:
Done